add quarel demo

joelgrus opened the pull request:
for the most part this is a straight port of Oyvind's existing demo into the new refactored world.

there was an `appendDefault` piece in the original that I didn't entirely understand, but I think it was trying to be more general and that the correponding behavior here is correct.

also, the trained quarel model referenced a trained tagger model that used a deprecated parameter, so I had to update the tagger mode and then the quarel model. this code reflects that too.

-- 

there is a section at the end where the fine-grained NER model is commented out, it needs to be similarly updated, but downloading and uploading these models is a pain on home network, so I'll fix that tomorrow, and I won't merge until that's done. in the meantime, that has nothing to do with the quarel demo itself, so that can still be reviewed.

schmmd commented on demo/src/components/demos/QuarelZero.js:
```diff
@@ -0,0 +1,179 @@
+]
+
+const ActionInfo = ({ action, question_tokens }) => {
+    const question_attention = action['question_attention'].map(x => [x])
```
What is this doing?

OyvindTafjord replied:
This piece is basically just copied from https://github.com/allenai/allennlp-demo/blob/master/demo/src/components/demos/WikiTables.js#L40

matt-gardner commented:
I'm pretty sure the `appendDefault` bit is important - it's what puts the rest of the default qualitative relations into the model.  If you look [here](http://aristo-demos-adhoc.dev.ai2/quarel-q-parsing-zero) and select the first example, the example in the code only has the sugar/diabetes relation, but when you select it you get the whole list.  It's cumbersome to add the whole list every time.  An easy way around this is to just have those qualitative relations hard-coded in every example (possibly all saved in a class variable to make that easier).  At least, I'm pretty sure that's what that's doing.  @OyvindTafjord might have more to say, though.

OyvindTafjord commented on demo/src/components/demos/QuarelZero.js:
```diff
@@ -0,0 +1,189 @@
+  Answer story questions about qualitative relations (QuaRel dataset) while adding new relations
+  without retraining. This uses the QuaSP+Zero semantic parser described in <i>QuaRel: A Dataset and Models for
+  Answering Questions about Qualitative Relationships</i> (in submission). The first few examples use
+  new relations, the rest are from the validation set.
```
Could we change "in submission" to "AAAI 2019", linked to https://arxiv.org/abs/1811.08048  (or http://data.allenai.org/quarel/)?

OyvindTafjord commented:
One small comment about the paper reference in the intro. Also note that the tagger model which the parser depends on will not be cached inside the docker image by the `cache_models.py` script which might be annoying for deployment purposes?

joelgrus commented:
ok, I think this should be good, I made the following changes

* fixed the "appendDefault" behavior, I think it's doing the right thing now
* changed the description as requested
* fixed the fine-grained-ner model and added it back

it doesn't look like the dockerfile is actually running cache_models right now (I think @schmmd took that out because our docker images are too big), but when that script runs it not only downloads but hydrates each model, which means it should also download transitively required models.
